### PR TITLE
docs: update README.md

### LINKED ISSUE #53372: README has some grammar problems
## Enhancement

Nothing too big, just some grammar improvements that I thought would help with the getting started experience.

### PR DESCRIPTION
**Pull request by DanRoscigno**
Some grammar edits

### What problem does this PR solve?

Issue Number: close #53372

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [x] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lilin90 — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -61,13 +61,13 @@ TiDB Cloud is the fully-managed service of TiDB, currently available on AWS and
...
 
-See [TiDB Cloud Quick Start Guide](https://docs.pingcap.com/tidbcloud/tidb-cloud-quickstart).
+See the [TiDB Cloud Quick Start Guide](https://docs.pingcap.com/tidbcloud/tidb-cloud-quickstart).
 
 ### Start with TiDB
 
 See [TiDB Quick Start Guide](https://docs.pingcap.com/tidb/stable/quick-start-with-tidb).
 
-### Start developing TiDB
+### Start developing with TiDB
```
Comment (lance6716) [inline README.md:70]: we mean "contribute to tidb's codebase" rather than "use tidb to develop your own project"

**Review by ngaut — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (lilin90) [reply README.md:70]: For the sentence we put in line 72, it should be "Start developing TiDB" here. Perhaps we should update the description in line 72. For contributing to TiDB's code, it's also covered in the following "[Contributing](https://github.com/pingcap/tidb/blob/master/README.md#contributing)" section.

File: README.md [hunk 2]
```diff
@@ -61,13 +61,13 @@ TiDB Cloud is the fully-managed service of TiDB, currently available on AWS and
...
 
 See the [Get Started](https://pingcap.github.io/tidb-dev-guide/get-started/introduction.html) chapter of [TiDB Development Guide](https://pingcap.github.io/tidb-dev-guide/index.html).
```
Comment (lilin90) [inline README.md:72]: How about this?
```suggestion
See [TiDB Developer Guide](https://docs.pingcap.com/tidb/stable/dev-guide-overview) and [TiDB Cloud Developer Guide](https://docs.pingcap.com/tidbcloud/dev-guide-overview).
```

File: README.md [hunk 3]
```diff
@@ -61,15 +61,15 @@ TiDB Cloud is the fully-managed service of TiDB, currently available on AWS and
...
 
-See [TiDB Cloud Quick Start Guide](https://docs.pingcap.com/tidbcloud/tidb-cloud-quickstart).
+See the [TiDB Cloud Quick Start Guide](https://docs.pingcap.com/tidbcloud/tidb-cloud-quickstart).
 
 ### Start with TiDB
 
 See [TiDB Quick Start Guide](https://docs.pingcap.com/tidb/stable/quick-start-with-tidb).
 
-### Start developing TiDB
+### Start developing with TiDB
```
Comment (DanRoscigno) [inline README.md:70]:
```suggestion
### Contributing to TiDB development
```

File: README.md (hunk 1 above)
Comment (DanRoscigno) [reply README.md]: I changed it to this: 
```
### Contributing to TiDB development
```
I think this is the clearest way to write this.

**Review by lance6716 — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (lilin90) [reply README.md]: @DanRoscigno I see that you accepted my suggestion via https://github.com/pingcap/tidb/pull/53373/commits/edcfa38d9f015927cf729ccb079283cebfefc1f4. IMO, with this suggestion, this title "Start developing with TiDB" is more appropriate.

**Pull request merged**